llvm/lib/Target/X86/lab3/simonyan_suren/X86SimonyanMICounter.cpp

```diff
@@ -0,0 +1,45 @@
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86Subtarget.h"
+#include "llvm/CodeGen/MachineFunctionPass.h"
+#include "llvm/CodeGen/MachineInstrBuilder.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+
+using namespace llvm;
+
+namespace {
+
+class X86SimonyanMICounterPass : public MachineFunctionPass {
+public:
+  static char ID;
+
+  X86SimonyanMICounterPass() : MachineFunctionPass(ID) {}
+
+  bool runOnMachineFunction(MachineFunction &MF) override {
+    const TargetInstrInfo *TII = MF.getSubtarget().getInstrInfo();
+    DebugLoc DL3 = MF.front().begin()->getDebugLoc();
+
+    for (auto &MBB : MF) {
+      unsigned count = 0;
+      for (auto &MI : MBB) {
+        if (!MI.isDebugInstr())
+          ++count;
+      }
+
+      BuildMI(MBB, MBB.getFirstTerminator(), DL3, TII->get(X86::ADD64ri32))
+          .addImm(count)
+          .addExternalSymbol("ic");
```

Comment:
External symbol refers to external functions or intrinsics. Please use another method for `ic`.

Reply:
I'm sorry I do not understand you. **ic** is an external symbol, so I used this method. Maybe you meant that it is a global variable and needs to be done through the appropriate method for the global variable?

Reply:
Sorry, missed your question.
Yes, I meant that `ic` is a global variable, whilst external symbol method is for calling functions defined in another translation units to be linked into current one later.

Reply:
comments were taken into account, I made changes